incubator/model-core/src/main/java/io/aklivity/zilla/runtime/model/core/internal/Int32Format.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.model.core.internal;
+
+import org.agrona.DirectBuffer;
+import org.agrona.collections.MutableInteger;
+
+public enum Int32Format
+{
+
+    TEXT
+    {
+        @Override
+        public int decode(
+            MutableInteger decoded,
+            MutableInteger processed,
```

Comment:
Suggest we add a separate `Int32State` class and use instead of these two mutable integers.
```
public final class Int32State
{
    public int decoded;
    public int processed;
}
```
then pass an instance here as a parameter called `state`, and refer to the fields as needed below.